server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala

```diff
@@ -134,6 +136,12 @@ class WebSocketRoutes @Inject() (@Named(InjectionKeys.RFC_8621) val authenticato
                 StateChangeListener(pushEnable.dataTypes.getOrElse(typeStateFactory.all.toSet), clientContext.outbound),
                 AccountIdRegistrationKey.of(clientContext.session.getUser)))
               .doOnNext(newRegistration => clientContext.withRegistration(newRegistration))
+              .`then`(SFlux.fromPublisher(delegationStore.delegatedUsers(clientContext.session.getUser))
+                .flatMap(delegatedUser => eventBus.register(
+                  StateChangeListener(pushEnable.dataTypes.getOrElse(typeStateFactory.all.toSet), clientContext.outbound),
+                  AccountIdRegistrationKey.of(delegatedUser)))
+                .doOnNext(newRegistration => clientContext.withRegistration(newRegistration))
```

Comment:
I am unsure about unregistering the old registration here, could need a bit more attention.

Reply:
It get's tricky because If bob delegates account for alice then alice needs to have 2 registration: alice one and bob one.


We need client context to hold a list of registration, somehow.